espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/Source.java

```diff
@@ -0,0 +1,100 @@
+package io.appium.espressoserver.lib.handlers;
+
+import android.view.View;
+import android.view.ViewGroup;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import java.io.StringWriter;
+
+import javax.annotation.Nullable;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import io.appium.espressoserver.lib.handlers.exceptions.AppiumException;
+import io.appium.espressoserver.lib.model.AppiumParams;
+import io.appium.espressoserver.lib.model.ViewAttributesEnum;
+import io.appium.espressoserver.lib.model.ViewElement;
+import io.appium.espressoserver.lib.viewaction.RootViewFinder;
+
+public class Source implements RequestHandler<AppiumParams, String> {
+
+    @Override
+    @Nullable
+    public String handle(AppiumParams params) throws AppiumException {
+
+        try {
+            // Build a document
+            Document doc;
+            DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder docBuilder = docFactory.newDocumentBuilder();
+            doc = docBuilder.newDocument();
+
+            // Populate the XML
+            View rootView = (new RootViewFinder()).getRootView();
+            buildXML(doc, null, rootView);
+
+            // Write the XML to string
+            TransformerFactory transformerFactory = TransformerFactory.newInstance();
+            Transformer transformer = transformerFactory.newTransformer();
+            DOMSource source = new DOMSource(doc);
+            StringWriter stringWriter = new StringWriter();
+            StreamResult result = new StreamResult(stringWriter);
+            transformer.transform(source, result);
+            return stringWriter.toString();
+
+        } catch (ParserConfigurationException pe) {
+            throw new AppiumException(String.format("Could not parse XML from source: %s", pe.getMessage()));
```

Comment:
does it make a sense to keep the parent stacktrace by adding _cause_ argument to AppiumException constructor?